src/sagas/tokens.js

```diff
@@ -0,0 +1,246 @@
+import {
+  takeEvery,
+  select,
+  delay,
+  call,
+  fork,
+  take,
+  all,
+  put,
+} from 'redux-saga/effects';
+import { metadataApi } from '@hathor/wallet-lib';
+import { channel } from 'redux-saga';
+import { get } from 'lodash';
+import { TxHistory } from '../models';
+import { specificTypeAndPayload } from './helpers';
+import {
+  types,
+  tokenFetchBalanceSuccess,
+  tokenFetchBalanceFailed,
+  tokenFetchHistorySuccess,
+  tokenFetchHistoryFailed,
+} from '../actions';
+
+const CONCURRENT_FETCH_REQUESTS = 5;
+const METADATA_MAX_RETRIES = 0;
```

Comment:
We were not retrying before this PR, so I left this at 0, should we enable it? @pedroferreira1

Reply:
Yes it is already implemented

Updated it to 3 in ad1eb4a9cd302848085a7d08380f25cc81324e2a